unlock-app/src/hooks/usePaywallLocks.ts

```diff
@@ -17,13 +17,13 @@ export const usePaywallLocks = (lockAddresses: string[]) => {
     })
 
     const locks = await Promise.all(lockPromises)
-    setLocks(locks)
     setLoading(false)
+    setLocks(locks)
   }
 
   useEffect(() => {
     getLocks()
-  }, [lockAddresses])
+  }, [])
```

Comment:
This was running infinitely before. Since the config is statically in the URL and is not expected to change, it's ok to make this `useEffect` run once and only once.